server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -240,6 +352,35 @@ private void writeDefaultValue(StreamOutput out, Object defaultValue) throws IOE
         }
     }
 
+    private Object readParser(StreamInput in, Object parser) throws IOException {
+        switch (type) {
+            case Boolean:
+                return new BooleanParser(in);
+            case Integer:
+                return new IntegerParser(in);
+            case Long:
+                return new LongParser(in);
+            case Float:
+                return new FloatParser(in);
+            case Double:
+                return new DoubleParser(in);
+            case TimeValue:
+                if (in.readBoolean()) {
+                    return new MinMaxTimeValueParser(in);
+                } else {
+                    return new MinTimeValueParser(in);
+                }
+            case ByteSizeValue:
+                if (in.readBoolean()) {
+                    return new ByteSizeValueParser(in);
+                } else {
+                    return new MemorySizeValueParser(in);
+                }
+            default:
+                throw new IllegalArgumentException("A SettingType has been added to the enum and not handled here.");
+        }
+    }
+
```

Comment:
How are we testing WriteableSetting?

Reply:
I think we do not need to add more test for `writeableSetting` class, the writeable parser class has been test in `SettingTests` class, what we did for writeableSetting is to update these parser in createSetting method to make sure that when customer pass new changes in custom setting, those custom setting will be checked by these parser.